AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -62,8 +63,13 @@ @interface ASDataController () {
 
   BOOL _delegateDidInsertNodes;
   BOOL _delegateDidDeleteNodes;
+  BOOL _delegateDidMoveNode;
   BOOL _delegateDidInsertSections;
   BOOL _delegateDidDeleteSections;
+  
+  NSMutableArray *_moveFromIndexPaths;
+  NSMutableArray *_moveToIndexPaths;
+  NSMutableDictionary *_moveToNodeContextsDict;
```

Comment:
Let's add type annotations and documentation here. What are these used for, what queue are the modified on, etc.